500.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * The template for displaying 500 pages (internal server errors)
+ *
+ * @link https://github.com/xwp/pwa-wp#offline--500-error-handling
+ *
+ * @package wp_rig
+ */
+
+namespace WP_Rig\WP_Rig;
+
+get_header(); ?>
+
+	<main id="primary" class="site-main">
+
+		<section class="error-500 not-found">
+			<header class="page-header">
+				<h1 class="page-title"><?php esc_html_e( 'Oops! Something went wrong.', 'wp-rig' ); ?></h1>
+			</header><!-- .page-header -->
+
+			<div class="page-content">
+				<p><?php esc_html_e( 'Something prevented the page from being rendered. Please try again.', 'wp-rig' ); ?></p>
```

Comment:
This should use `wp_service_worker_error_message_placeholder()` here instead. See: https://github.com/xwp/twentyseventeen-westonson/blob/03934b7328f9b18dd979ae97458ad38feb6f0e1a/500.php#L25-L32

Reply:
@westonruter Ah thanks, so `wp_service_worker_error_message_placeholder()` prints a general message, while `wp_service_worker_error_details_template()` is only for showing the actual cause of the 500?

Which markup does `wp_service_worker_error_message_placeholder()` create? Just a paragraph tag with a sentence like this, or something else? Is this customizable?

Also, what happens if the site is on production and should not display the actual fatal error message for security reasons (via `ini_set( 'display_errors', 0 )`? Will `wp_service_worker_error_details_template()` automatically detect that and not output anything then?

Reply:
> Ah thanks, so `wp_service_worker_error_message_placeholder()` prints a general message, while `wp_service_worker_error_details_template()` is only for showing the actual cause of the 500?

Correct. See https://github.com/xwp/pwa-wp/blob/be8c1d702b4cf99a577b67012d09533702d4f70e/wp-includes/js/service-worker-navigation-routing.js#L67-L92

> Which markup does `wp_service_worker_error_message_placeholder()` create? Just a paragraph tag with a sentence like this, or something else? Is this customizable?

It creates a paragraph tag with a comment inside. See https://github.com/xwp/pwa-wp/blob/733563f46996ea2d5aaf52a576971f92c3748164/wp-includes/service-workers.php#L433-L438

It's not intended to be customizable.

> Also, what happens if the site is on production and should not display the actual fatal error message for security reasons (via `ini_set( 'display_errors', 0 )`? Will `wp_service_worker_error_details_template()` automatically detect that and not output anything then?

In that case, nothing will be displayed. Correct: https://github.com/xwp/pwa-wp/blob/be8c1d702b4cf99a577b67012d09533702d4f70e/wp-includes/js/service-worker-navigation-routing.js#L71-L73